src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -68,11 +68,6 @@ void ReportSchedulerImpl::OnReadHandlerCreated(ReadHandler * aReadHandler)
                     "Registered a ReadHandler that will schedule a report between system Timestamp: %" PRIu64
                     " and system Timestamp %" PRIu64 ".",
                     newNode->GetMinTimestamp().count(), newNode->GetMaxTimestamp().count());
-
-    Milliseconds32 newTimeout;
-    // No need to check for error here, since the node is already in the list otherwise we would have Died
-    CalculateNextReportTimeout(newTimeout, newNode);
-    ScheduleReport(newTimeout, newNode);
```

Comment:
Hmm.  So the node will still be in the list, though, with possibly garbage values for its timeouts?  And other nodes ending up with scheduling events will be affected by those garbage values?

It seems like we should maybe move away from registering on "handler created" and instead register on "subscription established".  We have a callback at that point anyway, which is currently piggybacking on the "finished a report" callback... but conceptually _that's_ when we should start paying attention to a read handler right?  When the priming read is done.

Reply:
There is no "scheduled" report expected before OnSubscriptionEstablished, because the scheduling is not known until then.  All that's expected is the priming read.

Did we move responsibility for _those_, along with the handling of reads (not subscribes) into the scheduler?  If so, that's a bit weird, because those need to happen "immediately", not scheduled.  But also: with the ScheduleReport call there removed, what actually makes the priming read happen?

Reply:
The priming read happened on the ForceDirtyState, which would trigger a report since at that point the intervals had not been updated from 0.

This however only worked due to the check of IsReportable() in Engine::Run(), which treated read reports like subscription reports with max interval of 0. 

I separated those in the read handler and the Engine, now we should only schedule subscription reports.